### PR TITLE
[CALCITE] FIRST_VALUE/LAST_VALUE with IGNORE/RESPECT NULLS

### PR DESCRIPTION
**Pull request by AndrewPochapsky**
- Added AST node SqlFirstLastValue
   - This doesn't have a field for the window because that is unparsed in a somewhat special way
- Added parsing logic
- Added tests

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/SqlFirstLastValue.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  @Override public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
+    if (first) {
+      writer.keyword("FIRST_VALUE");
+    } else {
+      writer.keyword("LAST_VALUE");
+    }
+    SqlWriter.Frame frame = writer.startList("(", ")");
+    value.unparse(writer, leftPrec, rightPrec);
+    if (nullOption != null) {
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/SqlFirstLastValue.java:69]: nit: Why not use an UNSPECIFIED enum value here to be more explicit?

File: core/src/main/java/org/apache/calcite/sql/SqlFirstLastValue.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+   * @param first  Whether or not this is the FIRST_VALUE or LAST_VALUE
+   * @param value  The value which is being used
+   * @param nullOption  If the nulls should be ignored or respected
+   */
+  public SqlFirstLastValue(SqlParserPos pos, boolean first, SqlNode value,
+      NullOption nullOption) {
+    super(pos);
+    this.first = first;
+    this.value = Objects.requireNonNull(value);
+    this.nullOption = nullOption;
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/SqlFirstLastValue.java:50]: Objects.requireNonNull()?

File: core/src/main/java/org/apache/calcite/sql/SqlKind.java [hunk 3]
```diff
@@ -547,6 +547,9 @@
    */
   VALUES,
 
+  /** {@code FIRST_LAST_VALUE} expression. */
+  FIRST_LAST_VALUE,
```
Comment (ArthurInTheShell) [inline core/src/main/java/org/apache/calcite/sql/SqlKind.java:551]: I am not sure if it is necessary for this expression to have its own SqlKind, would it be in either SqlKind.OTHER or SqlKind.OTHER_FUNCTION? Could you provide more context on why FIRST_LAST_VALUE should have it's own SqlKind?

Comment (AndrewPochapsky) [reply core/src/main/java/org/apache/calcite/sql/SqlKind.java:551]: Having a specific SqlKind may be preferable by downstream consumers. What do you think @dasch-1 ?

Comment (dasch-1) [reply core/src/main/java/org/apache/calcite/sql/SqlKind.java:551]: Yes, having a distinct SqlKind is always preferred by downstream consumers.

**Pull request merged**